subsys/net/ip/ipv6_fragment.c

```diff
@@ -365,39 +365,62 @@ void net_ipv6_frag_foreach(net_ipv6_frag_cb_t cb, void *user_data)
 }
```

Comment:
:+1: to the `net: ipv6_fragment: improve reassembly condition`, great fix. I just wonder if you could add unit tests to `tests/net/ipv6_fragment/src/main.c` to catch these issues.
Having unit tests for all the commits in this PR might be too much work, but we could have some tests at least for this commit. The fixes in this PR are great but they assume that the conformance tests are run periodically to catch any regressions in the IPv6 code. This assumption might be true but if we have at least some tests implemented under `tests/net/`, we could catch any regression(s) earlier.

Reply:
I will look into implementing some unit tests